messaging/kafka_topics.py

```diff
@@ -0,0 +1,265 @@
+
+# Create and update multiple topics at once (faster!)
+- kafka_topics:
+    topics: [my-topic, my-other-topic]
+    partitions: 1
+    replicas: 1
+    zookeeper: localhost:2181
+
+# Create and update multiple topics with specific partitions and replicas
+- kafka_topics:
+    topics:
+      - name: my-topic
+        partitions: 2
+        replicas: 1
+      - name: my-other-topic
+        partitions: 1
+        replicas: 3
+    zookeeper: localhost:2181
+
+# Mix overrides and defaults
+- kafka_topics:
+    topics:
+      - my-topic
+      - name: my-other-topic
+      - name: my-third-topic
+        partitions: 12
+        replicas: 3
+    partitions: 1
+    replicas: 2
+    zookeeper: localhost:2181
+"""
+
+import re
+import collections
+
+# Initialiize Ansible Module
+# import module snippets
+from ansible.module_utils.basic import *
+
+def read_topics(kafka_topics, m):
+    p = m.params
+
+    cmd = "%s --zookeeper %s --describe" % (kafka_topics, p["zookeeper"])
+    # If a single topic was given, it will be faster to only check it
+    if p["topic"]:
+        cmd += " --topic %s" % (p["topic"],)
+
+    rc, out, err = m.run_command(cmd)
+    if rc:
+        m.fail_json(msg="listing topics failed: %s" % (err), stdout=out)
+
+    topics = {}
+    config_re = re.compile("^Topic:(?P<name>\S+)\s*"
+                           "PartitionCount:(?P<partitions>\d+)\s*"
+                           "ReplicationFactor:(?P<replicas>\d+)",
+                           flags=re.MULTILINE)
+    matches = config_re.finditer(out)
+    for match in matches:
+        topics[match.group("name")] = {
+                "partitions": int(match.group("partitions")),
+                "replicas": int(match.group("replicas")),
+                }
+
+    return topics
+
+
+def execute_change(kafka_topics, m, state, topic, partitions, replicas):
+    p = m.params
+
+    cmd = "%s --zookeeper %s --%s --topic %s --partitions %d" % (kafka_topics,
+            p["zookeeper"], state, topic, partitions)
+
+    if state == "create":
+        cmd += " --replication-factor %d" % (replicas,)
+
+    rc, out, err = m.run_command(cmd)
+    if rc:
+        m.fail_json(msg="%s of topic '%s' failed: %s" % (state, topic, err),
+                stdout=out)
+
+
+def main():
+    """
+    Main execution block and ansible boiler plate
+    """
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            path=dict(default=None),
+            zookeeper=dict(required=True),
+            topics=dict(default=None, type='list'),
+            topic=dict(default=None, aliases=["name"]),
+            partitions=dict(default=None, type="int"),
+            replicas=dict(default=None, type="int"),
+        ),
+        required_one_of=[["topic", "topics"]],
+        supports_check_mode=True,
+    )
+
+    changed = False
+
+    p = module.params
+    args = ["partitions", "replicas"]
+
+    topics = {}
+    if p["topic"]:
```

Comment:
Do you mean the arguments to the `AnsibleModule` constructor, or something else? I couldn't figure out how to do what I wanted withe the constructor, but I can look again.